### PR TITLE
[Lesson 02] Created 2 jobs and APIs for them.

### PR DESCRIPTION
**Pull request by denys-ops**
I've created 2 jobs (as required) and APIs for them using FastAPI (instead of Flask). 
Both APIs use same main.py to start.
The logic could be verified by `check_jobs.py`.
Also covered my function by unit tests.

### CODE REVIEW

File: lesson_02/main.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    server = uvicorn.Server(config)
+    await server.serve()
+
+
+async def main():
+    # Створюємо завдання для кожного сервера
+    task1 = asyncio.create_task(run_server(get_app, "127.0.0.1", 8081))
+    task2 = asyncio.create_task(run_server(transform_app, "127.0.0.1", 8082))
+
+    # Виконуємо завдання одночасно
```
Comment (mikolaz27) [inline lesson_02/main.py:33]: use english comments and this no need to comment this part, usually use comments for some complicated part of code

File: lesson_02/main.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+async def run_server(app, host, port):
+    config = uvicorn.Config(app=app, host=host, port=port)
+    server = uvicorn.Server(config)
+    await server.serve()
+
+
+async def main():
+    # Створюємо завдання для кожного сервера
+    task1 = asyncio.create_task(run_server(get_app, "127.0.0.1", 8081))
```
Comment (mikolaz27) [inline lesson_02/main.py:30]: could be moved to settings

File: lesson_02/apis/transform_api/service.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import json
+import os
+from fastavro import writer, parse_schema
+avro_schema = {
```
Comment (mikolaz27) [inline lesson_02/apis/transform_api/service.py:4]: its good to have it in separate module or settings.

File: lesson_02/apis/collect_api/service.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    """Collect data using API from all pages and save them into files."""
+    # Create or Re-create RAW dir
+    if not os.path.exists(raw_dir):
+        os.makedirs(raw_dir, exist_ok=True)
+    else:
+        shutil.rmtree(raw_dir, ignore_errors=True)
+        os.makedirs(raw_dir)
+
+    # Save the data
+    for page in range(1, 3001):
```
Comment (mikolaz27) [inline lesson_02/apis/collect_api/service.py:22]: what if count of pages will be different? Try to use while.

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+/*/.venv/
+.idea
```
Comment (mikolaz27) [inline .gitignore:2]: In pycharm you can generate ignore file. It will contain all the required ignores

image

Comment (denys-ops) [reply .gitignore:2]: For some reason I don't have such option in my list
Знімок екрана 2024-04-10 о 22 21 46

File: lesson_02/main.py (hunk 1 above)
Comment (denys-ops) [reply lesson_02/main.py]: I used to create a skeleton with a comments before implementation. Just forgot to delete them. Fixed.

File: lesson_02/main.py (hunk 2 above)
Comment (denys-ops) [reply lesson_02/main.py]: Make sense, updated

File: lesson_02/apis/collect_api/service.py (hunk 4 above)
Comment (denys-ops) [reply lesson_02/apis/collect_api/service.py]: Sounds reasonable ;)

**Pull request merged**